be/src/runtime/stream_load/transaction_mgr.cpp

```diff
@@ -397,30 +397,29 @@ void TransactionMgr::_clean_stream_context() {
         auto ctx = _exec_env->stream_context_mgr()->get(id);
         if (ctx != nullptr) {
             int64_t now = UnixSeconds();
-            // try lock fail means transaction in processing
-            if (ctx->lock.try_lock()) {
-                // abort timeout transaction
-                if ((now - ctx->begin_txn_ts) > ctx->timeout_second && ctx->timeout_second > 0) {
-                    ctx->status = Status::Aborted(fmt::format("transaction is aborted by timeout."));
-                    auto st = _rollback_transaction(ctx);
-                    LOG(INFO) << "Abort transaction " << ctx->brief() << " since timeout " << ctx->timeout_second
-                              << " begin ts " << ctx->begin_txn_ts << " status " << st;
-                }
-
-                if (ctx->body_sink != nullptr) {
-                    if (!ctx->body_sink->exhausted()) {
+            Status status;
+            if ((now - ctx->begin_txn_ts) > ctx->timeout_second && ctx->timeout_second > 0) {
```

Comment:
```suggestion
            if (ctx->txn_timed_out(now)) {
```
let StreamLoadContext handle the detailed time out logic